Sources/PackageLoading/ManifestSignatureParser.swift

```diff
@@ -0,0 +1,177 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+@_implementationOnly import struct Foundation.Data
+
+import TSCBasic
+
+public enum ManifestSignatureParser {
+    public static func parse(manifestPath: AbsolutePath, fileSystem: FileSystem) throws -> ManifestSignature? {
+        let manifestContents: ByteString
+        do {
+            manifestContents = try fileSystem.readFileContents(manifestPath)
```

Comment:
we have a helper in Basics that returns Data instead of ByteString (which we want to deprecate). Maybe easier to use tat instead